apps/DisplayCluster/resources/MultiTouchGestureArea.qml

```diff
@@ -0,0 +1,113 @@
+import QtQuick 2.0
+import DisplayCluster 1.0
+import "qrc:/qml/core/style.js" as Style
+
+MultiPointTouchArea {
+
+    signal tapAndHold(real x, real y)
+    signal tapStarted(real x, real y)
+    signal tapEnded()
+    signal pan(real x, real y, real dx, real dy)
+    signal panEnded()
+    signal doubleTap()
+
+    anchors.fill: parent
+
+    minimumTouchPoints: 0
+    maximumTouchPoints: 1
+    mouseEnabled: true
+
+    touchPoints: TouchPoint { id: p0; }
+
+    property bool panning: false
+    property bool secondRelease: false
+    property int numberOfTap: 0
+
+    property real startX: 0
+    property real startY: 0
+    property real previousOffsetX: 0
+    property real previousOffsetY: 0
+    // Needed because of QTBUG-44370. The "release" mouse event is sent 2 times.
+    property bool pressed: false
+
+    property QtObject obj2: Timer {
+        id: tapAndHoldTimer
+        interval: Style.tapAndHoldTiming
+        running: false
+        repeat: false
+        onTriggered: {
+            if(manhattanLengthCheck())
+                tapAndHold(p0.x, p0.y)
+        }
+    }
+    property QtObject obj: Timer {
+        id: doubleTapTimer
+        interval: Style.doubleTapTiming
+        running: false
+        repeat: false
+        onTriggered: {
+            numberOfTap = 0
+        }
+    }
+
+    function manhattanLengthCheck()
+    {
+        return Math.abs(p0.startX - p0.x) + Math.abs(p0.startY - p0.y) < Style.touchPixelDelta
+    }
+
+    onPressed: {
+        pressed = true
+
+        startX = p0.x
+        startY = p0.y
+
+        tapStarted(p0.x, p0.y)
+        tapAndHoldTimer.restart()
+        if(numberOfTap == 0)
+            doubleTapTimer.restart()
+
+        numberOfTap += 1
+    }
+    onReleased: {
+        if(!pressed)
+            return
+        pressed = false
+        tapAndHoldTimer.stop()
+
+        if(numberOfTap == 2)
+        {
+            doubleTapTimer.stop()
+            numberOfTap = 0
+            doubleTap()
+        }
+        else
+        {
+            if(panning)
+            {
+                previousOffsetX = 0
+                previousOffsetY = 0
+                panEnded()
+            }
+            else
+                tapEnded()
+        }
+        panning = false
+        previousOffsetX = 0
+        previousOffsetY = 0
+    }
+    onUpdated: {
+        panning = true
```

Comment:
The old recognizer for tap is different. if the manhattanLength of the delta is less than 40, it's still a tap, not a pan. On the wall that is crucial because having a perfect tap with not even a single pixel movement is almost impossible :)